geode-core/src/distributedTest/java/org/apache/geode/internal/cache/partitioned/BucketCreationCrashCompletesRegressionTest.java

```diff
@@ -95,13 +95,62 @@ public void setUp() {
    */
   @Test
   public void testCrashWhileCreatingABucket() {
-    vm1.invoke(() -> createPartitionedRegionWithObserver());
-    vm2.invoke(() -> createPartitionedRegionWithObserver());
+    for (VM vm : toArray(vm1, vm2)) {
+      vm.invoke(() -> {
+        DistributionMessageObserver.setInstance(new CrashMemberBeforeManageBucketMessage(vm0));
 
-    vm0.invoke(() -> createAccessorAndCrash());
+        PartitionAttributesFactory<?, ?> partitionAttributesFactory =
+            new PartitionAttributesFactory();
+        partitionAttributesFactory.setRedundantCopies(1);
+        partitionAttributesFactory.setRecoveryDelay(0);
 
-    vm1.invoke(() -> verifyBucketsAfterAccessorCrashes());
-    vm2.invoke(() -> verifyBucketsAfterAccessorCrashes());
+        RegionFactory<?, ?> regionFactory = getCache().createRegionFactory(PARTITION);
+        regionFactory.setPartitionAttributes(partitionAttributesFactory.create());
+
+        regionFactory.create(regionName);
+      });
+    }
+
+    vm0.invoke(() -> {
+      PartitionAttributesFactory<?, ?> partitionAttributesFactory =
+          new PartitionAttributesFactory<>();
+      partitionAttributesFactory.setRedundantCopies(1);
+      partitionAttributesFactory.setLocalMaxMemory(0);
+
+      RegionFactory<String, String> regionFactory = getCache().createRegionFactory(PARTITION);
+      regionFactory.setPartitionAttributes(partitionAttributesFactory.create());
+
+      Region<String, String> region = regionFactory.create(regionName);
+
+      // trigger the creation of a bucket, which should trigger the destruction of this VM.
+      assertThatThrownBy(() -> region.put("ping", "pong")).isInstanceOf(CancelException.class);
+    });
+
+    for (VM vm : toArray(vm1, vm2)) {
+      vm.invoke(() -> {
+        PartitionedRegion pr = (PartitionedRegion) getCache().getRegion(regionName);
+        int totalNumBuckets = pr.getAttributes().getPartitionAttributes().getTotalNumBuckets();
+        for (int i = 0; i < totalNumBuckets; i++) {
+          int bucketId = i;
+
+          await().until(() -> {
+            try {
+              return pr.getBucketOwnersForValidation(bucketId) != null;
+            } catch (ForceReattemptException e) {
+              return false;
+            }
+          });
+
+          if (pr.getBucketOwnersForValidation(bucketId).isEmpty()) {
+            continue;
+          }
+
+          await().untilAsserted(() -> {
+            assertThat(pr.getBucketOwnersForValidation(bucketId)).hasSize(2);
```

Comment:
could be expression lambda (there's 3 others, but I'm good with the readability as is)